src/zarr/store/memory.py

```diff
@@ -47,6 +50,7 @@ async def exists(self, key: str) -> bool:
         return key in self._store_dict
 
     async def set(self, key: str, value: Buffer, byte_range: tuple[int, int] | None = None) -> None:
+        self._check_writable()
```

Comment:
It doesn't seem easy for a method like `set` or `set_partial_values` to handle the exception themselves, since the exception gets raised in `_check_writable()`. Also, if someone wants to check if the store is writeable, but _not_ raise an exception (e.g., when including the writable state in a repr of the store), then that would also be hard with `_check_writable()`.

What if `self.writable` was a boolean, and we did `if not self.writable: raise BlaBlaBlaException()` in methods like `set`? This would solve the two issues I raised above.

Reply:
@d-v-b - I'm not sure I follow. We can certainly add a `writable` property but wouldn't the behavior be the same?

Reply:
Yes, with a `writeable` property, the overall behavior of `set` would be the same (raise an exception if `writeable` is False), but the `writeable` property would also be accessible in contexts when we don't want to raise an exception, e.g. displaying a repr for the store that includes the writeable mode. The current design (`_check_writeable()`) preemptively raises an exception and makes this second use case cumbersome.

Reply:
... but we can sort this particular issue out at a later time